src/gui/qgscodeeditor.cpp

```diff
@@ -90,16 +92,35 @@ void QgsCodeEditor::keyPressEvent( QKeyEvent *event )
 
 void QgsCodeEditor::setSciWidget()
 {
+  QHash< QString, QColor > colors;
+  if ( QgsApplication::instance()->themeName() != QStringLiteral( "default" ) )
+  {
+    QSettings ini( QgsApplication::instance()->uiThemes().value( QgsApplication::instance()->themeName() ) + "/qscintilla.ini", QSettings::IniFormat );
+    for ( const auto &key : ini.allKeys() )
+    {
+      colors.insert( key, QgsSymbolLayerUtils::decodeColor( ini.value( key ).toString() ) );
+    }
+  }
+
   setUtf8( true );
   setCaretLineVisible( true );
-  setCaretLineBackgroundColor( QColor( 252, 243, 237 ) );
+  setCaretLineBackgroundColor( colors.value( QStringLiteral( "caretLineColor" ), QColor( "#fcf3ed" ) ) );
```

Comment:
Can you change the string literals for the default color to the numeric QColor constructor instead? It's more efficient then the string parser.